File: frontend/src/api/apply.ts
import { http } from './http'

export interface ApplyResultOut {
  job_id: string
  title: string
  status: string
  message: string
}

export interface ApplyTaskOut {
  id: number
  credential_id: number | null
  credential_username: string
  status: string
  total_count: number
  success_count: number
  failed_count: number
  skipped_count: number
  results: ApplyResultOut[]
  start_time: string | null
  end_time: string | null
  error_message: string | null
  created_at: string
}

export const applyApi = {
  create: (data: { credential_id: number; job_ids?: string[] | null }) =>
    http.post<ApplyTaskOut>('/apply', data).then((r) => r.data),
  list: () => http.get<ApplyTaskOut[]>('/apply').then((r) => r.data),
  get: (id: number) => http.get<ApplyTaskOut>(`/apply/${id}`).then((r) => r.data),
  remove: (id: number) => http.delete(`/apply/${id}`),
}

File: backend/app/schemas/apply.py
from datetime import datetime

from pydantic import BaseModel


class ApplyTaskCreate(BaseModel):
    credential_id: int
    job_ids: list[str] | None = None


class ApplyResultOut(BaseModel):
    job_id: str
    title: str
    status: str
    message: str = ""


class ApplyTaskOut(BaseModel):
    id: int
    credential_id: int | None
    credential_username: str
    status: str
    total_count: int
    success_count: int
    failed_count: int
    skipped_count: int
    results: list[ApplyResultOut]
    start_time: datetime | None
    end_time: datetime | None
    error_message: str | None
    created_at: datetime

    model_config = {"from_attributes": True}
